table/block_based_table_reader.cc

```diff
@@ -1352,6 +1368,22 @@ Status BlockBasedTable::PutDataBlockToCache(
     assert(raw_block_contents->is_raw_block);
 #endif  // NDEBUG
 
+    // Block fetcher always use memory allocator from regular block cache.
+    // Copy the content to compressed block cache.
+    //
+    // If there's a way to know if the block to read from file is compressed
+    // or not, we can save this memory copy by using the right memory
+    // allocator in BlockFetcher.
+    if (memory_allocator != memory_allocator_compressed) {
+      size_t block_size = raw_block_contents->data.size();
+      size_t alloc_size = block_size + kBlockTrailerSize;
```

Comment:
@siying Shall I include the trailer when copying to compressed block cache, or shall I remove it?

Reply:
Yes you always need to.